### ISSUE TITLE
Enhancement: naming-convention ignore imported types

### ISSUE THREAD

**Issue by ghosttie** [typescript-eslint/typescript-eslint#6744]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention/

### Description

I have no control over the naming conventions of imported types, so the naming-convention rule that I set up for my code is throwing errors everywhere I interact with imported code that needs me to pass it an object e.g. the jQuery `ajax` function.

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

**Comment by bradzacher** [comment]
> I have no control over the naming conventions of imported types

That's not strictly true - you can alias imported names:

```ts
import { ajax as Ajax } from 'jquery';
```

---

Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2106

**Comment by ghosttie** [comment]
That's not what I mean - I mean the ajax function needs me to pass an object to it:

```ts
$.ajax({
  type: 'GET',
  url: '/something/something',
  data: {
    something: something
  },
  dataType: 'json',
  success: function() {
    // do something
  }
});
```

and all of those property names (`type`, `url`, `data` etc.) are all underlined in red because they don't match my naming rule.

But if I change them to match my naming rule the jQuery won't recognize what I sent it.

**Comment by bradzacher** [comment]
Going back to my previous point... You included no code samples and described a different scenario.

Again - this is why we have an issue template with required fields - so you can clearly describe the problem with clear examples to remove any misunderstanding and back-and-forth.

---

Regardless, duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2244
